bootstrap-app/src/cmr/bootstrap/data/metadata_retrieval/metadata_transformer.clj

```diff
@@ -0,0 +1,108 @@
+  [context concept _ target-formats]
+  (let [{concept-mime-type :format, metadata :metadata, concept-type :concept-type} concept
+        source-version (umm-spec/umm-json-version concept-type concept-mime-type)
+        [t result] (u/time-execution (reduce (fn [translated-map target-format]
+                                               (assoc translated-map target-format
+                                                      (umm-json/umm->json
+                                                       (u/remove-nils-empty-maps-seqs
+                                                        (vm/migrate-umm context
+                                                                        concept-type
+                                                                        source-version
+                                                                        (umm-spec/umm-json-version concept-type
+                                                                                                   target-format)
+                                                                        (json/decode metadata true))))))
+                                             {}
+                                             target-formats))]
+    (info "transform-with-strategy migrate-umm-json: "
+          "time: " t
+          "concept-mime-type: " concept-mime-type
+          "concept-type: " concept-type
+          "parent request num-concepts: " (:num-concepts concept)
+          "target-formats: " target-formats
+          "source version: " source-version
+          "provider: " (:provider-id concept)
+          "metadata length: " (count metadata))
+    result))
+
+(defn transform-with-strategy
+  "Depending on the transformation strategy pick the correct function to call to translate
+  the concept to the target format."
+  [context concept strategy target-formats]
+  (case strategy
+    :current-format
+    {(mt/mime-type->format (:format concept))
+     (cx/remove-xml-processing-instructions (:metadata concept))}
+    
+    :migrate-umm-json
+    (transform-with-strategy-migrate-umm-json context concept strategy target-formats)
+
+    :umm-spec
+    (transform-with-strategy-umm-spec context concept strategy target-formats)
+    
+    (errors/internal-error!
+     (format "Unexpected transform strategy [%s] from concept of type [%s] to [%s]"
+             strategy (:format concept) (pr-str target-formats)))))
+
+(defn transform-to-multiple-formats
+  "Transforms the concept into multiple different formats. Returns a map of target format to metadata."
+  [context concept target-formats ignore-exceptions?]
+  {:pre [(not (:deleted concept))]}
+  (->> target-formats
+       (group-by #(transform-strategy concept %))
+       (keep (fn [[k v]]
+               (if ignore-exceptions?
+                 (try
+                   (transform-with-strategy context concept k v)
+                   (catch Throwable e
+                     (log/error
```

Comment:
is this a warning if you are ignoring it?

Reply:
Not sure, this was copied from the original code.  Its to ignore the exception when running a test.